### DISCUSSION TITLE
Displaying Read-Only Attribute

### DISCUSSION THREAD

**Discussion by firewalker06** [excid3/madmin#175] [General]
Since madmin works by iterating through `model.attributes`, I was thinking using `ActiveModel::Attributes::ClassMethods#attribute` would work if we want to have a Read-Only attribute in madmin. Turns out its true:

```ruby
# models/invoice.rb
class Invoice < ApplicationRecord
  attribute :total

  def total
    # do some calculations
  end
end

# madmin/resources/invoice_resource.rb
class InvoiceResource < Madmin::Resource
  attribute :total, form: false # mocking as read-only
end
```

But I think it would be good idea to have `read_attribute` that displays in Index and Show view.

**Comment by firewalker06** [comment]
Or a `read_only: true` option in madmin's `attribute` class method.

**Comment by excid3** [comment]
We could probably check for `attr_readonly` and default `form: false` for those.

```ruby
class Post < ActiveRecord::Base
  attr_readonly :title
end

post = Post.create!(title: "Introducing Ruby on Rails!")
post.update(title: "a different title") # change to title will be ignored
```
https://api.rubyonrails.org/classes/ActiveRecord/ReadonlyAttributes/ClassMethods.html
